src/action.rs

```diff
@@ -1,7 +1,21 @@
-//! The different actions that can be done.
+//! The different actions that can be executed via any given key.
 
 use crate::key_code::KeyCode;
 
+/// The different types of actions we support for key macros
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum SequenceEvent {
+    /// A keypress/keydown
+    Press(KeyCode),
+    /// Key release/keyup
+    Release(KeyCode),
+    /// Combination quick keydown followed by keyrelease
+    Tap(KeyCode),
+    /// Release all (currently) pressed keys
+    ReleaseAll(),
```

Comment:
Let's keep it simple first, and only support Press and Release

Reply:
Removed.  That was easy =)